### PR TITLE
Bump Harbor package to 2.4.2

### LINKED ISSUE #2731: Bump Harbor package to 2.4.x
## Feature Request

We need to bump Harbor package to 2.4.x (initially to 2.4.0)

### LINKED ISSUE #2942: harbor package 2.4.x support caBundleSecretName for chartmuseum, core, jobservice, registry, trivy components
## Feature Request

It would be nice to expose `caBundleSecretName`, which must contain key named "ca.crt", to the end users of harbor package so that they could inject the custom ca for core, jobservice, registry, trivy components.

### PR DESCRIPTION
**Pull request by zyyw**
Signed-off-by: Shengwen Yu <yshengwen@vmware.com>

## What this PR does / why we need it

Bump Harbor package to 2.4.2

## Which issue(s) this PR fixes

Fixes: #2731 
Fixes: #2942 
## Describe testing done for PR

1. ut
`cd addons/packages/harbor/2.4.2/test`
`make test`
2. functional test for tracing which was introduced in Harbor 2.4.2

## Special notes for your reviewer

In release note, please add 
```
Bump Harbor package to 2.4.2
```

### CODE REVIEW

**Comment by MinerYang** [comment]
lgtm

**Review by MinerYang — DISMISSED** [review]

**Comment by jorgemoralespou** [comment]
Have these issues been addressed along with the version bump?
- https://github.com/vmware-tanzu/community-edition/issues/3591
- https://github.com/vmware-tanzu/community-edition/issues/3491
- https://github.com/vmware-tanzu/community-edition/issues/2942
- https://github.com/vmware-tanzu/community-edition/issues/1994

**Review by seemiller — COMMENTED** [review]
Along with Jorge, I too am curious if this release addresses any outstanding Harbor package issues.

**Comment by zyyw** [comment]
Hi @jorgemoralespou / @seemiller , 
These issues are addressed:
* https://github.com/vmware-tanzu/community-edition/issues/3491 is updated in the issue (nothing to be address from the codebase perspective)
* https://github.com/vmware-tanzu/community-edition/issues/2942  is addressed in this 2.4.2 version
* https://github.com/vmware-tanzu/community-edition/issues/1994 is addressed in this PR https://github.com/vmware-tanzu/community-edition/pull/2563, also included in 2.4.2 version

This one is not:
* https://github.com/vmware-tanzu/community-edition/issues/3591

Please let me know if you have further concerns.

File: addons/packages/harbor/2.4.2/test/e2e/harbor_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+	}
+
+	By("installing harbor addon package")
+	harborHostname = fmt.Sprintf("harbor.%s.nip.io", getReachableContourEnvoyIP())
+
+	harborAdminPassword = generateHarborPassword()
+
+	packageName := utils.TanzuPackageName("harbor")
+
+	version := findPackageAvailableVersion(packageName, "2.3.3")
```
Comment (seemiller) [inline addons/packages/harbor/2.4.2/test/e2e/harbor_suite_test.go:124]:
```suggestion
	version := findPackageAvailableVersion(packageName, "2.4.2")
```

I assume that this should be `2.4.2` for the `2.4.2` version of the package?

**Review by seemiller — CHANGES_REQUESTED** [review]

**Review by MinerYang — DISMISSED** [review]
lgtm

**Comment by heww** [comment]
Hi @zyyw I think https://github.com/vmware-tanzu/community-edition/issues/3591 must be fixed in this PR, it's a P0 issue.

**Review by seemiller — APPROVED** [review]

**Pull request merged**